examples/_quick-start/java/Example.java

```diff
@@ -0,0 +1,57 @@
+package com.hashicorp.quickstart;
+
+import java.util.Map;
+import java.util.HashMap;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.support.Versioned;
+import org.springframework.vault.core.VaultTemplate;
+
+@SpringBootApplication
+public class Example implements CommandLineRunner {
+    public static void main(String[] args) {
+        ConfigurableApplicationContext context = SpringApplication.run(Example.class, args);
+        context.close();
+    }
+
+    @Override
+    public void run(String... strings) throws Exception {
+        VaultEndpoint vaultEndpoint = new VaultEndpoint();
+
+        vaultEndpoint.setHost("127.0.0.1");
+        vaultEndpoint.setPort(8200);
+        vaultEndpoint.setScheme("http");
+
+        // Authenticate
+        VaultTemplate vaultTemplate = new VaultTemplate(vaultEndpoint,
+                new TokenAuthentication("dev-only-token"));
+
+        // Write a secret
+        Map<String, String> data = new HashMap<>();
+        data.put("password", "Hashi123");
+
+        Versioned.Metadata createResponse = vaultTemplate
+                .opsForVersionedKeyValue("secret")
+                .put("my-secret-password", data);
+
+        System.out.println("Secret written successfully.");
+
+        // Read a secret
+        Versioned<Map<String, Object>> readResponse = vaultTemplate
+                .opsForVersionedKeyValue("secret")
+                .get("my-secret-password");
+
+        String password = readResponse.getData().get("password").toString();
```

Comment:
We should protect against NPE here as a best practice. Let's identify if any of the objects in these cascading calls could be null and add checks accordingly

Reply:
Would it be better to just catch the exception (or even let it escape) since this is just a quick start?

Reply:
We actually protected against these in all our Go examples but not the C# ones, so we're already kind of a mixed bag, haha... but let's see if there's a simple way to do it.

Looks like we have this in the hello-vault-spring app today:
```
        assert response != null;
        if ( response.hasData() ) {
            String apiKey = (String) Objects.requireNonNull(response.getData()).get(apiKeyField);
        }
```

Would doing that same thing be enough here?

Reply:
Changed as discussed with Vinny and Anton